node/router/src/heartbeat.rs

```diff
@@ -117,32 +161,15 @@ pub trait Heartbeat<N: Network>: Outbound<N> {
             return;
         }
 
-        // Retrieve the trusted peers.
-        let trusted = self.router().trusted_peers();
-        // Retrieve the bootstrap peers.
-        let bootstrap = self.router().bootstrap_peers();
-
-        // Find the oldest connected peer, that is neither trusted nor a bootstrap peer.
-        let oldest_peer = self
-            .router()
-            .get_connected_peers()
-            .iter()
-            .filter(|peer| !trusted.contains(&peer.ip()) && !bootstrap.contains(&peer.ip()))
-            .filter(|peer| !self.router().cache.contains_inbound_block_request(&peer.ip())) // Skip if the peer is syncing.
-            .filter(|peer| self.is_block_synced() || self.router().cache.num_outbound_block_requests(&peer.ip()) == 0) // Skip if you are syncing from this peer.
-            .min_by_key(|peer| peer.last_seen())
-            .map(|peer| peer.ip());
-
         // Disconnect from the oldest connected peer, if one exists.
-        if let Some(oldest) = oldest_peer {
+        if let Some(oldest) = self.get_removable_peers().pop().map(|p| p.ip()) {
             info!("Disconnecting from '{oldest}' (periodic refresh of peers)");
             let _ = self.send(oldest, Message::Disconnect(DisconnectReason::PeerRefresh.into()));
             // Disconnect from this peer.
             self.router().disconnect(oldest);
         }
     }
 
-    /// TODO (howardwu): If the node is a validator, keep the validator.
```

Comment:
I think this has been addressed, but let me know if the TODO should be kept.

Reply:
The list returned by `get_removable_peers` still contains validators.

The only code I see really keeping validators is under the branch `if self.router().node_type().is_prover()`. I think we change that with minimal side-effects to `if self.router().node_type().is_prover || self.router().node_type().is_validator()` to that.

Reply:
This was more or less my question. I don't think we should keep all validators, otherwise the set of peers could get very big. The behavior now is to prioritize keeping them over clients, but if that causes problems we can change it.